fabric-rendering-v1/src/client/java/net/fabricmc/fabric/mixin/client/rendering/LayeredDrawerMixin.java

```diff
@@ -0,0 +1,42 @@
+/*
+ * Copyright (c) 2016, 2017, 2018, 2019 FabricMC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package net.fabricmc.fabric.mixin.client.rendering;
+
+import java.util.List;
+import java.util.function.BooleanSupplier;
+
+import org.spongepowered.asm.mixin.Final;
+import org.spongepowered.asm.mixin.Mixin;
+import org.spongepowered.asm.mixin.Shadow;
+import org.spongepowered.asm.mixin.injection.At;
+import org.spongepowered.asm.mixin.injection.Inject;
+import org.spongepowered.asm.mixin.injection.callback.CallbackInfoReturnable;
+
+import net.minecraft.client.gui.LayeredDrawer;
+
+@Mixin(LayeredDrawer.class)
+public abstract class LayeredDrawerMixin {
+	@Shadow
+	@Final
+	private List<LayeredDrawer.Layer> layers;
+
+	@Inject(method = "addSubDrawer", at = @At("HEAD"), cancellable = true)
+	private void flattenSubDrawer(LayeredDrawer drawer, BooleanSupplier shouldRender, CallbackInfoReturnable<LayeredDrawer> cir) {
+		layers.addAll(((LayeredDrawerAccessor) drawer).getLayers());
```

Comment:
No,  we should flattern them I designed the wrapper around them being layerd, this also breaks vanilla as the `shouldRender` is never checked.

We only need to wrap the `drawer` here so we can get the sub layers in the visitor. I would go back to having the sublayer record I had before (without the id) but creating it here.

Reply:
Ah, I misunderstood, done.